### DISCUSSION TITLE
Script to add note from annotations?

### DISCUSSION THREAD

**Discussion by spencer-michaels** [windingwind/zotero-actions-tags#219] [Q&A]
I desperately want to set up a way to add a note from annotations automatically (be it via a tag or shortcut), but I don't know how to write scripts for this plugin. Does anyone have something similar to what I'm looking for? Thanks so much! <3

**Comment by oliverkuehle** [comment]
Hey, I quickly created myself a script a while back. 
It first deletes all existing Extracted Annotation notes and then creates a new one. It does this for each item that is selected at the time the script runs.

- If you set the action event to 'Close Tab', then it will always update your annotation note after you close the tab/pdf-viewer.
- Alternatively, you can add and run the action using the menus, which enabled you to bulk update multiple items. For example, you can select (highlight) your whole library and then run the action from the menu, which will update the _Extracted Annotation_ notes for the whole library.

```
const items = await ZoteroPane.getSelectedItems()
for (item of items) {
    const noteIDs = item.getNotes();
    for (let id of noteIDs) {
        let note = Zotero.Items.get(id);
        let noteHTML = note.getNote();
        if (noteHTML.startsWith('<div data-citation-items') && noteHTML.includes('Annotations')) {
            Zotero.Items.trashTx(id);   
        }
    }       
}
ZoteroPane.addNoteFromAnnotationsFromSelected();
```

_Note: If you're using Zotero in a different language than english, you will have to replace the 'Annotations' in line 8 with whatever title stand at the start of your extracted annotation notes._

**Be aware that this works for me, but might not work for you or might have become deprecated by the time you read this. Use at your own risk, as there is no guarantee that it will work correctly.**

**Comment by zpofrp** [comment]
i try to use this, but tell me
script error: redeclaration of formal parameter items.
